bentoml/adapters/multi_image_input.py

```diff
@@ -39,61 +50,110 @@ class MultiImageInput(BaseInputAdapter):
         ImportError: imageio package is required to use ImageInput
 
     Example usage:
-        ```
-        class MyService(BentoService):
-            @bentoml.api(input=MultiImageInput(input_names=('imageX', 'imageY')))
-            def predict(self, image_groups):
-                for image_group in image_groups:
-                    image_array_x = image_group['imageX']
-                    image_array_y = image_group['imageY']
-        ```
+
+    >>> class MyService(BentoService):
+    >>> @bentoml.api(input=MultiImageInput(input_names=('imageX', 'imageY')))
+    >>> def predict(self, image_groups):
+    >>>     for image_group in image_groups:
+    >>>         image_array_x = image_group['imageX']
+    >>>         image_array_y = image_group['imageY']
     """
 
     def __init__(
         self,
         input_names=("image",),
-        accept_image_formats=None,
+        accepted_image_formats=None,
         pilmode="RGB",
         is_batch_input=False,
         **base_kwargs,
     ):
         if is_batch_input:
-            raise ValueError('ImageInput can not accpept batch inputs')
+            raise ValueError('ImageInput can not accept batch inputs')
         super(MultiImageInput, self).__init__(
             is_batch_input=is_batch_input, **base_kwargs
         )
         self.input_names = input_names
         self.pilmode = pilmode
-        self.accept_image_formats = (
-            accept_image_formats or get_default_accept_image_formats()
+        self.accepted_image_formats = (
+            accepted_image_formats or get_default_accept_image_formats()
         )
 
-    def handle_batch_request(self, requests, func):
-        """Handles an HTTP request, convert it into corresponding data
-        format that user API function is expecting, and return API
-        function result as the HTTP response to client
-
-        :param requests: List of flask request object
-        :param func: user API function
-        """
-        raise NotImplementedError
+    def handle_request(self, request: Request, func):
+        files = {
+            name: self.read_file(file.filename, file.stream)
+            for (name, file) in request.files.items()
+        }
+        result = func((files,))[0]
+        return self.output_adapter.to_response(result, request)
+
+    def read_file(self, name: str, file: BytesIO):
+        safe_name = secure_filename(name)
+        verify_image_format_or_raise(safe_name, self.accepted_image_formats)
+        return imread(file, pilmode=self.pilmode)
+
+    def handle_batch_request(
+        self, requests: Iterable[SimpleRequest], func
+    ) -> Iterable[SimpleResponse]:
+        inputs = []
+        slices = []
+        for i, req in enumerate(requests):
+            content_type = req.headers['Content-Type']
+
+            if "multipart/form-data" not in content_type:
```

Comment:
> https://github.com/bentoml/BentoML/blob/master/bentoml/adapters/image_input.py#L180
> Better to catch known exceptions, otherwise one request with invalid input will break other requests in the same batch.

This is handled here. Neither the `ImageInput` nor the `MultiImageInput` handle the file body being an invalid image payload (possible if someone manages to submit an arbitrary file masquerading as a valid image file)